snap/quota/quota.go

```diff
@@ -364,17 +435,47 @@ func contains(s []int, e int) bool {
 	return false
 }
 
-// validateCPUsAllowedResourceFit locates the nearest parent group that has a cpu-set quota, and then verifies
+// validateCPUsAllowedResourceFit verifies that the new cpu-set doesn't conflict with the current reserved cpu-set
+// of the group, and if not locates the nearest parent group that has a cpu-set quota, and then verifies
 // that the requested cpu cores match a subset of the previously set allowance.
 func (grp *Group) validateCPUsAllowedResourceFit(allQuotas map[string]*groupQuotaAllocations, cpusAllowed []int) error {
+
+	// isSuperset returns true if a is a superset of b.
+	isSuperset := func(a, b []int) bool {
+		for _, b1 := range b {
```

Comment:
Suggestion, as first quick check before the loop you could return false immediately if `len(a) < len(b)` (unless it's never the case).

Reply:
Ah nice suggestion, I'll include that in another PR!